pkg/reconciler/revision/resources/queue.go

```diff
@@ -191,13 +226,21 @@ func makeQueueContainer(rev *v1alpha1.Revision, loggingConfig *logging.Config, o
 		volumeMounts = append(volumeMounts, internalVolumeMount)
 	}
 
+	rp := rev.Spec.GetContainer().ReadinessProbe.DeepCopy()
+
+	applyReadinessProbeDefaults(rp, userPort)
+
+	// TODO(joshrider) bubble up error instead of squashing it here
+	probeJSON, _ := readiness.EncodeProbe(rp)
+
 	return &corev1.Container{
 		Name:            QueueContainerName,
 		Image:           deploymentConfig.QueueSidecarImage,
 		Resources:       createQueueResources(rev.GetAnnotations(), rev.Spec.GetContainer()),
 		Ports:           ports,
-		ReadinessProbe:  queueReadinessProbe,
+		ReadinessProbe:  makeQueueProbe(rp),
 		VolumeMounts:    volumeMounts,
+		Args:            []string{"--readiness-probe", probeJSON},
```

Comment:
Why not pass this as an environment variable?

Reply:
Seems like an option. Is an environment variable preferred here?

Reply:
Everything else is an env-variable, so this is the oddball. I'd adjust for consistency.